chap-cheri-riscv.tex

```diff
@@ -536,39 +474,6 @@ \subsubsection{Capability Exception Reporting}
 general-purpose capability register file.
 \end{description}
 
-\subsubsection{Capability Control and Status Registers (CCSRs)}
-\label{subsubsec-ccsrs}
-New per HART \xccsr{} \texttt{XLEN}-bit RISC-V CSRs are defined as per
-Figure~\ref{fig-ccsr} (shown for XLEN=32):
-
-\begin{figure}[!h]
-\begin{center}
-\begin{bytefield}[bitwidth=\textwidth/34]{32}
-  \bitheader[endianness=big]{0,1,2,31} \\
-  \bitbox{30}{\textbf{WIRI}}
-  \bitbox{1}{\texttt{d}}
-  \bitbox{1}{\texttt{e}}
-\end{bytefield}
-\caption{\xccsr{} register format; WIRI bits are Write Ignore Read Ignore.}
-\label{fig-ccsr}
-\end{center}
-\end{figure}
-
-\pmnote{WIRI has been removed from the RISC-V specification.  The spec
-  now typically uses WPRI (Write-Preserve, Read-Ignore) for such
-  undefined bits (e.g. in \texttt{mstatus}).  Both WARL (Write-Any,
-  Read-Legal) and WLRL (Write-Legal, Read-Legal) seem unsuitable here.
-  The RISC-V spec unsatisfyingly does not specify what happens if
-  software doesn't preserve values for WPRI, but then this spec
-  doesn't need to either.}
-
-\begin{description}
-\item [e] The \texttt{e} ``enable'' bit \ajnote{read only? / only wriFable in u-mode? only writable with a specific permission in pcc?} tells whether capability extensions are enabled or disabled.
```

Comment:
I think keeping the enabled bit makes sense. I agree that dirty is not that useful.

Reply:
Jess noted that if all we want is per-mode enable bits then the more idiomatic approach on RISC-V now would be to allocate new bits in `Xstateen` via the now-ratified `Sstateen` extension instead of using custom CSRs.  Given that we haven't yet implemented the enable bit, perhaps it would be best to add that as a replacement instead?

Reply:
That sounds good to me.

Reply:
Hmmm, if we stored non-address bits in the lower word we would also still need `CSub`?

Reply:
(Assuming we don't want to follow the TBI model where code is aware of TBI and has to mask out the bits explicitly from addresses)

Reply:
Yes. I continue to believe putting bits that aren't exposed to C as the address in the lower word is a really bad idea for all these kinds of reasons.

Reply:
(i.e. people are free to investigate these if they want to, but we shouldn't concern ourselves with all that weirdness in the main CHERI-RISC-V spec outside of some appendix rambling to act as prior art)

Reply:
I've added a sentence to both `CGetAddr` and `CSub`.  I'm not sure how you wouldn't avoid the need for a `CSub` if you stored the version in the address similar to TBI but wanted to avoid explicit masking.  Were you assuming you'd do a normal `Sub` between two results of `CGetAddr`?  Presumably we wouldn't have needed `CSub` at all even in the split register case if that was reasonably efficient?

Reply:
I guess within the same object it cancels out, and subtraction between objects is UB...